src/geojson-layer.ts

```diff
@@ -50,8 +64,11 @@ export default class GeoJSONLayer extends React.Component<Props, void> {
     const layerId = `${id}-${type}`;
     layerIds.push(layerId);
 
-    const paint: Paints = this.props[`${type}Paint`] || {};
-    const layout: Layouts = this.props[`${type}Layout`] || {};
+    const paint: Paints = this.props[`${typeToLayerLUT[type]}Paint` ] || {};
+
+    // default undefined layers to invisible
+    const visibility = Object.keys(paint).length ? 'visible' : 'none';
```

Comment:
Hey, having a second look at the PR and I am wondering, why do you default to invisible ? Can you convince me it is an expected behaviour by default ?